model/manifest.go

```diff
@@ -254,6 +254,7 @@ func FindManifest(dir string) (manifest *Manifest, path string, err error) {
 		f, ferr := os.Open(path)
 		if ferr != nil {
 			if !os.IsNotExist(ferr) {
+				path = ""
```

Comment:
Can you explain how this fixes the problem?

Reply:
Thanks @SmartHoneybee :)

Reply:
Ah, part of the underlying problem here is the named return parameters and breaking the pattern of returning `nil` values when returning an error. I wonder if we should refactor to make the returns explicit instead? [A Tour of Go](https://tour.golang.org/basics/7) claims:
> Naked return statements should be used only in short functions, as with the example shown here. They can harm readability in longer functions.

Reply:
Never been a huge fan of named return parameters, I'm all for the refactoring

Reply:
should we refactor in this PR or after?

Reply:
Thinking a bit more clearly today, I guess it's not the named return parameters that is the problem, but the naked return. There's a whole thread of discussions about the future of naked returns in Go 2: https://github.com/golang/go/issues/21291.

My suggestion would be to perhaps `return nil, "", ferr` instead of directly assigning to `path`, eliminating the naked return now, and deferring any potential and larger refactor until later.

Reply:
@lieut-data did the change, can you please take a look when you have time? thanks!

Reply:
Sorry for the ambiguity here, @cpanato, but I think this is only slightly better at the moment, since it still relies on setting `path` (and `err`) correctly in all the right places instead of returning explicitly the desired outcome.

The underlying problem with the naked return:
```
path = ""
err = ferr
return
```

is that it relies on `path` and `err` being set correctly -- as you spotted and fixed in this issue. However, changing the code to be:
```
path = ""
err = ferr
return nil, path, err
```

is only slightly better, since the risk of not setting the return values of `path` or `err` is still present (imagine a copy and paste of the `return` that fails to copy the assigning parts). Best would be to just return explicitly:
```
return nil, "", ferr
```

Given the use of named parameters here, this is synactically equivalent to assigning to `path` and `err`, but much clearer and without any potential ambiguity.

My proposal would be to just `return nil, "", ferr`, or `return nil, "", ioerr"`, etc., throughout this method, simultaneously eliminating all the assignments to the named return values.